src/main/java/programminglife/model/Graph.java

```diff
@@ -54,71 +70,71 @@ private void parseLink(Scanner sc) {
-
         while (lineScanner.hasNextLine()) {
             String line = lineScanner.nextLine();
             Scanner tokenScanner = new Scanner(line);
             tokenScanner.useDelimiter("\t");
             String token = tokenScanner.next();
             numLinesRead++;
 
-            if (numLinesRead % 2500 == 0)
-                verbose = true;
+            if (verbose && numLinesRead % lineInterval == 0)
+                logCurrentLine = true;
             else
-                verbose = false;
+                logCurrentLine = false;
 
-            if (verbose) System.out.println(String.format("Token %s read (line %d)... ", token, numLinesRead));
+            if (logCurrentLine) System.out.println(String.format("Token %s read (line %d)...", token, numLinesRead));
 
             switch (token) {
                 case "S":
                     // Parse segment
-//                    System.out.println("parsing segment");
-                    Node parsedNode = Node.parseSegment(tokenScanner);
-                    Node existingNode = graph.getNode(parsedNode.getId());
-                    if (existingNode == null)
-                        graph.addNode(parsedNode);
-                    else
-                        existingNode.setSequence(parsedNode.getSequence());
-
+                    graph.parseSegment(tokenScanner);
                     numNodesParsed++;
                     break;
                 case "L":
                     // Parse link
-//                    System.out.println("parsing link");
                     graph.parseLink(tokenScanner);
                     numLinksParsed++;
                     break;
                 default:
-//                    System.out.println("strange token on line!");
+                    // Otherwise
                     miscParsed++;
             }
             tokenScanner.close();
         }
 
         lineScanner.close();
 
-        System.out.println();
-        System.out.println(String.format("%d lines read from file %s", numLinesRead, file));
-        System.out.println(String.format("%d segments parsed", numNodesParsed));
-        System.out.println(String.format("%d links parsed", numLinksParsed));
-        System.out.println(String.format("%d miscellaneous parsed", miscParsed));
-        System.out.println();
+        if (verbose) System.out.println();
+        if (verbose) System.out.println(String.format("%d lines read from file %s", numLinesRead, file));
+        if (verbose) System.out.println(String.format("%d segments parsed", numNodesParsed));
+        if (verbose) System.out.println(String.format("%d links parsed", numLinksParsed));
+        if (verbose) System.out.println(String.format("%d miscellaneous parsed", miscParsed));
+        if (verbose) System.out.println();
 
-        for (Node n : graph.nodes) {
+        for (int id : graph.nodes.keySet()) {
+            Node n = graph.getNode(id);
             if (n != null && n.getParents().isEmpty() && n.getId() > 0) {
```

Comment:
id no longer has to be greater than 0 (as there should not be any ids lower than 0)